common/lib/xmodule/xmodule/tabs.py

```diff
@@ -68,14 +77,25 @@ def __init__(self, tab_dict):
         Args:
             tab_dict (dict) - a dictionary of parameters used to build the tab.
         """
-
+        super(CourseTab, self).__init__()
         self.name = tab_dict.get('name', self.title)
         self.tab_id = tab_dict.get('tab_id', getattr(self, 'tab_id', self.type))
-        self.link_func = tab_dict.get('link_func', link_reverse_func(self.view_name))
         self.course_staff_only = tab_dict.get('course_staff_only', False)
-
         self.is_hidden = tab_dict.get('is_hidden', False)
 
+        self.tab_dict = tab_dict
```

Comment:
The idea was to allow subclasses (and in particular the `TabFragmentViewMixin`) to be able to override the property. Maybe the others should be switched to be consistent, but they seem more static.